### ISSUE TITLE
Trouble in understanding regex syntax

### ISSUE THREAD

**Issue by gijo-varghese** [probablyup/buttermilk#7]
I'm trying to validate a route that will accept any routes like `http://localhost:3000/123456` where I can access `123456` as `props.params.some_id`

This is what I've tried:
```
      <Router
        routes={[
          {
            path: /^[0-9]$/,
            render: () => SomeComponent
          }
        ]}
      />
```
However the route is not matching.

**Comment by probablyup** [comment]
Okay so if you want the path parsed into a param, you need to do something like this instead of the regex:

`/:id`

**Comment by gijo-varghese** [comment]
That will accept any value given to id, right? It should be a number. That's why I need regex

**Comment by probablyup** [comment]
Correct. Parse it out using location.pathname then to grab the ID if you
specifically want to use regex. params is only set if you use the :name
syntax

**Comment by probablyup** [comment]
\d+ would probably be better for that regex also

**Comment by probablyup** [comment]
It’s also worth mentioning that regex matches the whole URL, so just write it to match against the end:

```js 
/\/\d+$/
```

**Comment by gijo-varghese** [comment]
So if I put a regex, I need to access it from `location.pathname`?

**Comment by probablyup** [comment]
Regex just matches the route for you and makes sure the right component is
rendered. The library then parses the URL into the various routing state
props like location and params (when available.)

From there it’s up to you what you do with that information.
location.pathname for that regex would be something like “/123456”, so you
could parseInt(location.pathname.slice(1), 10) to get the ID.

**Comment by gijo-varghese** [comment]
Ok. So that what should I put? 

```
<Router
        routes={[
          {
            path: /\/\d+$/
            render: () => SomeComponent
          }
        ]}
      />
```

??

**Comment by probablyup** [comment]
Yes and SomeComponent will have the routing props injected.

**Comment by gijo-varghese** [comment]
Well, its throwing an error: `TypeError: path.match is not a function`

**Comment by probablyup** [comment]
I’ll put together a codesandbox

**Comment by gijo-varghese** [comment]
ok

**Comment by probablyup** [comment]
Ok my apologies, there was a bug in the regex handling that I just fixed in 1.1.2. You can see a working example here now: https://codesandbox.io/s/93n39xz98y

**Comment by probablyup** [comment]
Really sorry for any time you lost on it - was missing some test cases :(

**Comment by probablyup** [comment]
@gijo-varghese Did that fix it for you?
